cylc/flow/remote.py

```diff
@@ -138,9 +138,30 @@ def run_cmd(
             return True
 
 
-def construct_ssh_cmd(raw_cmd, user=None, host=None, forward_x11=False,
-                      stdin=False, ssh_login_shell=None, ssh_cylc=None,
-                      set_UTC=False, allow_flag_opts=False):
+def construct_platform_ssh_cmd(raw_cmd, platform, **kwargs):
+    """A wrapper around `construct_ssh_cmd` allowing us to pass a platform
+    object rather than a user and host.
+
+    Args:
+        All as `construct_ssh_cmd` except for user and host.
+    """
+    ret = construct_ssh_cmd(
+        raw_cmd,
+        user=platform['owner'],
+        host=get_host_from_platform(platform),
+        ssh_cmd=platform['ssh command'],
+        ssh_cylc=platform['cylc executable'],
+        ssh_login_shell=platform['use login shell'],
+        **kwargs
+    )
+    return ret
```

Comment:
Bit cleaner?
```suggestion
    return construct_ssh_cmd(
        raw_cmd,
        user=platform['owner'],
        host=get_host_from_platform(platform),
        ssh_cmd=platform['ssh command'],
        ssh_cylc=platform['cylc executable'],
        ssh_login_shell=platform['use login shell'],
        **kwargs
    )
```

Reply:
It's what I originally did, but on reflection I prefer not to because the current format makes adding a debug statement easier if it were required.